src/xx-apt

```diff
@@ -88,13 +93,17 @@ fi
 fixubuntusources() {
   # fix all current sources to native arch
   nativearch="$(TARGETPLATFORM="" TARGETPAIR="" TARGETARCH="" TARGETOS="" xx-info arch)"
-  sed -E "/arch=/! s/^(# )?(deb|deb-src) /\1\2 [arch=$nativearch] /" -i "$(aptsourcesfile)"
+  if ! check_deb822; then
+    sed -E "/arch=/! s/^(# )?(deb|deb-src) /\1\2 [arch=$nativearch] /" -i "$(aptsourcesfile)"
+  else
+    sed -E "/Architectures:/! s/^(Types\: [a-z]+)/\1\nArchitectures: $nativearch/" -i "$(aptsourcesfile)"
+  fi
 
   if ! xx-info is-cross; then return; fi
 
   # add custom list for target
   debarch=$(xx-info debian-arch)
-  targetlist=/etc/apt/sources.list.d/xx-$debarch.list
+  targetlist=/etc/apt/sources.list.d/xx-$debarch
   if [ -f "$targetlist" ]; then return; fi
```

Comment:
This check seems to be broken with the filename change?

Reply:
True, corrected